File: src/GPS.WinTermConfig.UI/wwwroot/BlazacoJSInterop.js
window.Blazaco = window.Blazaco || {};
window.Blazaco.Editors = [];

window.Blazaco.Editor = {
    InitializeEditor: function (model) {
        const thisEditor = monaco.editor.create(document.getElementById(model.id), model.options);
        if (window.Blazaco.Editors.find(e => e.id === model.id)) {
            return false;
        } else {
            window.CurrentEditor = thisEditor;
            window.Blazaco.Editors.push({ id: model.id, editor: thisEditor });

            window.onresize = () => window.CurrentEditor.layout();
        }
        return true;
    },
    UpdateOptions: function(id, editorOptions) {
        const myEditor = window.Blazaco.Editors.find(e => e.id === id);
        if (!myEditor) {
            throw `Could not find a editor with id: '${window.Blazaco.Editors.length}' '${id}'`;
        }
        window.CurrentEditor = myEditor;
        window.CurrentEditor.editor.updateOptions(editorOptions);
        return true;
    },
    GetValue: function (id) {
        const myEditor = window.Blazaco.Editors.find(e => e.id === id);
        if (!myEditor) {
            throw `Could not find a editor with id: '${window.Blazaco.Editors.length}' '${id}'`;
        }
        window.CurrentEditor = myEditor;
        return window.CurrentEditor.editor.getValue();
    },
    SetValue: function (id, value) {
        const myEditor = window.Blazaco.Editors.find(e => e.id === id);
        if (!myEditor) {
            throw `Could not find a editor with id: '${window.Blazaco.Editors.length}' '${id}'`;
        }
        window.CurrentEditor = myEditor;
        window.CurrentEditor.editor.setValue(value);
        return true;
    },
    SetTheme: function (id, theme) {
        const myEditor = window.Blazaco.Editors.find(e => e.id === id);
        if (!myEditor) {
            throw `Could not find a editor with id: '${window.Blazaco.Editors.length}' '${id}'`;
        }
        window.CurrentEditor = myEditor;
        window.CurrentEditor.editor.setTheme(theme);
        return true;
    },
    Layout: function (id, layoutInfo) {
        const myEditor = window.Blazaco.Editors.find(e => e.id === id);
        if (!myEditor) {
            throw `Could not find a editor with id: '${window.Blazaco.Editors.length}' '${id}'`;
        }
        window.CurrentEditor = myEditor;
        window.CurrentEditor.editor.layout(layoutInfo);
        return true;
    }
};

File: src/GPS.WinTermConfig.UI/Editor/BlazacoJSInterop.cs
using Microsoft.JSInterop;
using System.Threading.Tasks;
using GPS.WinTermConfig.UI.Editor.Options;
using System;

namespace GPS.WinTermConfig.UI.Editor
{
    public static class BlazacoJSInterop
    {
        public static ValueTask<bool> InitializeEditor(IJSRuntime runtime, EditorModel editorModel)
            => runtime.InvokeAsync<bool>("Blazaco.Editor.InitializeEditor", new[] { editorModel });

        public static ValueTask<bool> UpdateOptions(IJSRuntime runtime, string id, EditorOptions editorOptions)
            => runtime.InvokeAsync<bool>("Blazaco.Editor.UpdateOptions", new object[] { id, editorOptions });

        public static ValueTask<string> GetValue(IJSRuntime runtime, string id)
            => runtime.InvokeAsync<string>("Blazaco.Editor.GetValue", new[] { id });

        public static ValueTask<bool> SetValue(IJSRuntime runtime, string id, string value)
            => runtime.InvokeAsync<bool>("Blazaco.Editor.SetValue", new[] { id, value });

        public static ValueTask<bool> SetTheme(IJSRuntime runtime, string id, string theme)
            => runtime.InvokeAsync<bool>("Blazaco.Editor.SetTheme", new[] { id, theme });
        public static ValueTask<bool> Layout(IJSRuntime runtime, string id, EditorLayoutInfo layoutInfo = null)
            => runtime.InvokeAsync<bool>("Blazaco.Editor.Layout", new object[] { id, layoutInfo });
    }
}
